Dockerfile

```
@@ -101,4 +101,9 @@ WORKDIR ${RUNNER_PATH}
 
 # COMMAND
 ENV IN_DOCKER=1
-CMD ["cml-runner"]
+# Smart entrypoint understands commands like `bash` or `/bin/sh` but defaults to `cml`;
+# also works for GitLab CI/CD
+# https://gitlab.com/gitlab-org/gitlab-runner/-/blob/4c42e96/shells/bash.go#L18-37
+# https://gitlab.com/gitlab-org/gitlab-runner/-/blob/4c42e96/shells/bash.go#L288
+ENTRYPOINT ["/bin/bash", "-c", "basename \"$0\" | grep -qE '^(b?a|z|fi|t?c)?sh$' && exec \"$0\" \"$@\" || exec cml \"$0\" \"$@\""]
```

fixes silly warning when doing e.g. `docker run ... --help`

```suggestion
ENTRYPOINT ["/bin/bash", "-c", "basename \"$0\" 2>/dev/null | grep -qE '^(b?a|z|fi|t?c)?sh$' && exec \"$0\" \"$@\" || exec echo cml \"$0\" \"$@\""]
```

---

I'd rather save a few bytes by doing it _The Right Way™_ 
```suggestion
ENTRYPOINT ["/bin/bash", "-c", "basename -- \"$0\" | grep -qE '^(b?a|z|fi|t?c)?sh$' && exec \"$0\" \"$@\" || exec cml \"$0\" \"$@\""]
```